tools/frr.service

```diff
@@ -20,4 +22,4 @@ ExecStop=/usr/lib/frr/frrinit.sh stop
 ExecReload=/usr/lib/frr/frrinit.sh reload
 
 [Install]
-WantedBy=network-online.target
+WantedBy=multi-user.target
```

Comment:
This doesn't really work for setups where IP addresses are configured in `zebra.conf` since other services may start up and be unable to bind to any configured IP / port since they haven't been added yet.

> This patch updates our service file so that it is properly started by the system when enabled, delayed until networking is up, and if possible delayed until after NetworkManager, systemd-networkd or any other networking configuration manager has finished performing its tasks - i.e. after network-online.target.

We probably need to order FRR parallel to these, not after them.  There isn't really any point in ordering FRR after them since additional config happening after we're started should be correctly picked up by zebra.  If anything we might even want to order _before_ these...

Reply:
(Actually this comment should've gone on the `After=network.target` line — it's not the `multi-user-target` that is the problem.)

`systemd-networkd.service` has this on my Debian box:
```
After=systemd-udevd.service network-pre.target systemd-sysusers.service systemd-sysctl.service
Before=network.target multi-user.target shutdown.target
```
I would say those should be correct for FRR too. (No idea what `systemd-sysusers.service` does though.)

Reply:
FWIW, `NetworkManager.service` is shorter:
```
Wants=network.target
After=network-pre.target dbus.service
Before=network.target
...
WantedBy=multi-user.target
```